.github/workflows/check-code.yml

```diff
@@ -218,69 +314,14 @@ jobs:
             ~/.cargo/registry/index/
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
-          key: ${{ github.job }}-${{ github.ref }}-${{ hashFiles('**/Cargo.lock') }}
+          key: ${{ github.job }}-${{ github.ref }}-${{ matrix.runtime }}-${{ hashFiles('**/Cargo.lock') }}
 
-      - name: Run `cargo doc`
+      # We don't care if the weight files are updated.
+      - name: Run runtime benchmarks
         run: |
           docker run --rm \
             -v "${GITHUB_WORKSPACE}:/workspace" \
             -v "${HOME}/.cargo:/root/.cargo" \
             -w /workspace \
-            -e RUSTDOCFLAGS='-D warnings' \
-            -e SKIP_WASM_BUILD=1 \
             paritytech/ci-unified:bullseye-1.74.0 \
-            bash -c "cargo doc --no-deps --locked ${{ matrix.features }}"
-
-  try-runtime:
-    name: Run try-runtime
-    runs-on: ubuntu-latest
-    needs: cargo-clippy
-    env:
-      TRY_RUNTIME_CLI_VERSION_TAG: v0.7.0
-    container:
-      image: paritytech/ci-unified:bullseye-1.74.0
-
-    strategy:
-      matrix:
-        runtime:
-          - peregrine
-          - spiritnet
-      fail-fast: false
-
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v4
-
-      - name: Free Disk Space
-        uses: jlumbroso/free-disk-space@main
-        with:
-          tool-cache: true
-
-      - name: Install try-runtime
-        run: |
-          curl -sL https://github.com/paritytech/try-runtime-cli/releases/download/${{ env.TRY_RUNTIME_CLI_VERSION_TAG }}/try-runtime-x86_64-unknown-linux-musl -o try-runtime
-          chmod +x ./try-runtime
-          ./try-runtime --version
-
-      - name: Set up Cargo cache
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-          key: ${{ github.job }}-${{ github.ref }}-${{ hashFiles('**/Cargo.lock') }}
-
-      - name: Build runtime
-        run: cargo build --release --locked -p ${{ matrix.runtime }}-runtime --features try-runtime
-
-      - name: Run `try-runtime`
-        run: |
-          ./try-runtime \
-          --runtime=./target/release/wbuild/${{ matrix.runtime }}-runtime/${{ matrix.runtime }}_runtime.compact.compressed.wasm \
-          on-runtime-upgrade \
-            --disable-spec-version-check \
-            --checks=all \
-          live \
-            --uri=wss://${{ matrix.runtime }}.kilt.io
+            bash -c "bash scripts/run_benches_for_runtime.sh ${{ matrix.runtime }}-dev dev > /dev/null"
```

Comment:
Not sure, if we would find errors with this approach. If one pallet is throwing an error, the script would continue with the next pallet, we needed to look up in the logs to find possible issues.

Reply:
You are right, the script was considering only the result of the last benchmarked pallet. I changed that [here](https://github.com/KILTprotocol/kilt-node/pull/731/commits/792ec806bc69707956e6ccef1e064e696afb0ae5), and made sure that we bail out early as long as one benchmark fails. I tested it by changing the benchmark helper of the switch pallet to make it fail (which it did), and also to try to run the `sudo` pallet for peregrine after it, but it did not because the script exited on the first failure.